### ISSUE TITLE
Create a new "Light" PDF template

### ISSUE THREAD

**Issue by alexmigf** [wpovernight/woocommerce-pdf-invoices-packing-slips#99]
@Spreeuw said:

> I also think it's time for an additional 'Light' template in PDF Invoices. It should have a few improvements over the old 'Simple' template such as:

1. Out of the box cross-compatibility with mpdf
2. Maybe using floats instead of columns for the addresses so that they can be moved to the other side more easily
3. No black header to save ink

**Comment by Spreeuw** [comment]
To document an idea we discussed some time ago:

I was thinking that we could also try a different sort of title/value setup, for example with small & big font like this:
![light-template-order-data-idea-1](https://user-images.githubusercontent.com/3217773/102689624-e6d33780-41ff-11eb-8131-5143cf909f1c.png)

One thing that might be difficult for this is that because of the hooks integration, this does need to be `table>tr>th+td`... so it needs to be all CSS but I think it's nice & fresh (and certainly not impossible to change `table-cell` to `block` ... at least in dompdf!)

**Comment by alexmigf** [comment]
What you think like this:

<img src="https://i.ibb.co/wsMnpVd/Captura-de-ecr-de-2021-01-12-15-55-01.png" alt="Captura-de-ecr-de-2021-01-12-15-55-01" border="0">

**Comment by Spreeuw** [comment]
very similar to my last mockup, but I'm still not very keen on the additional height this occupies... (yours takes a bit more than mine)

**Comment by alexmigf** [comment]
What about 3 columns, like we have now? order data, shipping and billing addresses, is the only way to avoid having a second row.

**Comment by dwalkerpriv** [comment]
@alexmigf are 4 columns too much?
![Screenshot 2022-09-09 102305](https://user-images.githubusercontent.com/62675729/189386218-04375730-7380-4968-a0a0-aa2b114c68ac.png)

**Comment by YordanSoares** [comment]
@alexmigf is this Light template intended to be part of our free main plugin, or is this a new premium template?

**Comment by alexmigf** [comment]
The original idea was for the free plugin, to help save ink.

**Comment by Spreeuw** [comment]
@YordanSoares both. Basically an alternative option in the free plugin, that will have a Premium counterpart too (just like the current Simple template).

**Comment by MohamadNateqi** [comment]
If I may, I wanted to mention that the column style suggested [here](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/issues/99#issuecomment-748469604), as Spreeuw noted, may not be optimal.

Considering the main plugin is more general, it would be better to keep the row style as it is more concise. However, we can offer the column style as a premium option so customers can choose the style that suits their invoice data.

By the way, I came across a column style that I think you might like:

![image](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/assets/136313810/d108dc9b-0deb-4596-932b-f7750c4b5d1a)
 
I would also like to suggest considering the Poppins font, as I found it quite nice.

I have another idea to share:

![image](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/assets/136313810/da231263-d6c0-41c5-8595-40eeb7d82baa)

**Comment by alexmigf** [comment]
I like the idea but having the order details spread in columns I'm not sure if it's flexible enough. Anyway, I think we should move from creating a new template from scratch to making the Simple template more light in terms of ink spending. We could call this template as Simple Light or Simple Lite, for example.

This comes from the other discussion we had, keeping in mind the popularity of the Simple and Simple Premium templates.

What you think?

**Comment by MohamadNateqi** [comment]
> What you think?

I totally agree, and I think it's better. The "Simple Lite" also is a good name.
Making a different structure and layout can be a job for another template.

**Comment by alexmigf** [comment]
Good! Do you want to take this?

**Comment by alexmigf** [comment]
Btw, the name can be other, was just something that came from my mind. Maybe we should discuss that with the team on Slack.

**Comment by MohamadNateqi** [comment]
> Do you want to take this?

Sure, good.

> Btw, the name can be other, was just something that came from my mind. Maybe we should discuss that with the team on Slack.

I found the "Simple Lite" quite good. But sure, we'll discuss it with the team on Slack.

**Comment by MohamadNateqi** [comment]
Since the purpose is to make the Simple template lighter, I created a sample and made changes only to the products table:

![image](https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/assets/136313810/401a0064-a406-43e5-82ac-255a3ee0fc67)

What do you think?

**Comment by dwalkerpriv** [comment]
@MohamadNateqi this looks pretty good. only thing that comes to mind as fine tuning is using darkgrey as color for the border-bottoms, except for the totals area. personally i'd make the logo smaller to simplify it more but that's just me. 

"Simple Lite" sounds to the point as well 👍

**Comment by MohamadNateqi** [comment]
@dwalkerpriv The dark gray color (#A9A9A9) doesn't meet WCAG contrast standards. I used a color (#595959) that passed all tests. I think this color is better for people with eye issues. What do you think?

Regarding the logo height, it's dynamic as it can change using the general settings.

**Comment by alexmigf** [comment]
Seems good for me.

**Comment by dwalkerpriv** [comment]
> The dark gray color (#A9A9A9) doesn't meet WCAG contrast standards. I used a color (#595959) that passed all tests. I think this color is better for people with eye issues. What do you think?

It's all good then 👍

**Comment by MohamadNateqi** [comment]
@alexmigf, considering that this template will only slightly modify the styles of the "Simple" template, all other files will remain untouched and copied from the "Simple" template.
Do you think it would be a good idea to implement the child-parent system of WordPress for themes in our template system?

**Comment by alexmigf** [comment]
Generally the php files are very identical, even for different templates. I think it doesn't worth having that work just for one template.

Maybe we could add something to distinguish from the Simple? Something customers ask a lot? What do you think?

**Comment by MohamadNateqi** [comment]
> Maybe we could add something to distinguish from the Simple? Something customers ask a lot? What do you think?

If there are any possible modifications to make on this template, which still keeps it the Simple Lite, then yeah, I think it would be great.
However, I don't know about customers' requests; can help with this?

**Comment by alexmigf** [comment]
@dwalkerpriv @YordanSoares do you think we could add some helpful for customers in the Lite template to distinguish from the Simple? Maybe something they ask a lot? Or any other ideas?

**Comment by alexmigf** [comment]
@MohamadNateqi in a discussion with  @Terminator-Barbapapa about this, I was thinking that probably we could add this Lite template without actually have any template files? Because we could use an hook to change the template styles, maybe we could make the code work to also have a "ghost" template? This way we avoid having to change things in the template structure.

But let me know all what you think about this, or if you have other suggestions/ideas.

**Comment by MohamadNateqi** [comment]
@alexmigf That's actually a great idea!
All other files except the style are the same right now, which was why I presented the child-parent system idea.

What do you think about adding a check box just for the Simple template below the template selection to enable the light mode? Then, the light CSS rules override and change the template. There is no need to add it to the template list too.

**Comment by alexmigf** [comment]
> What do you think about adding a check box just for the Simple template below the template selection to enable the light mode? 

That's even a better idea, I love it.

**Comment by Terminator-Barbapapa** [comment]
I do like the idea to make our templates more modular. Also when thinking about our new Customizer I think we need to make separate components to make that work. 

I would like to keep everything related to template selection in one setting for now though.

**Comment by MohamadNateqi** [comment]
> I would like to keep everything related to template selection in one setting for now though.

So you say we go for the "ghost" template, right?

**Comment by dwalkerpriv** [comment]
> @dwalkerpriv @YordanSoares @Terminator-Barbapapa do you think we could add some helpful for customers in the Lite template to distinguish from the Simple? Maybe something they ask a lot? Or any other ideas?

Probably in this case it would be more so what we can **take away** from the simple template to make the lite version more distinct.

Maybe reduce vertical space between table rows so more items could potentially hold on one page? Could look into reducing font sizes as well, or replacing <h> tags with bold fonts of the same size as the rest of the body font.

As far a addition goes, I can't really think of anything. Most requests besides saving ink is for layout changes. We will be addressing that with the customizer update.
